test_runner/regress/test_storage_controller.py

```diff
@@ -2091,3 +2091,44 @@ def test_storage_controller_step_down(neon_env_builder: NeonEnvBuilder):
         )
         == 0
     )
+
+def test_storage_controller_ps_restarted_during_drain(neon_env_builder: NeonEnvBuilder):
+    # single unsharded tenant, two locations
+    neon_env_builder.num_pageservers = 2
+
+    env = neon_env_builder.init_start()
+
+    # make sure there is no secondary, so we will have something to drain
+    env.storage_controller.tenant_policy_update(env.initial_tenant, {"placement": {"Attached": 1}})
+    env.storage_controller.reconcile_until_idle()
+
+    first, _ = env.pageservers
+
+    # attached_at = int(env.storage_controller.locate(env.initial_tenant)[0]["node_id"])
+    # secondary_at = next(filter(lambda ps: ps.id != attached_at, env.pageservers))
```

Comment:
does this test break if storage controller decided to put the attached on the second node?

Reply:
done in b0dcf3ddd